hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java

```diff
@@ -101,18 +120,18 @@ private static CGroupsHandler getInitializedCGroupsHandler(Configuration conf)
    */
 
   public static CGroupsHandler getCGroupsHandler() {
-    return cGroupsHandler;
+    return cgroupsV2Enabled ? cGroupsV2Handler : cGroupsV1Handler;
```

Comment:
getCGroupsHandler method is called from DefaultOOMHandler, CGroupElasticMemoryController, which seemingly rely on v1 memory limits, so the related changes should be addressed in [YARN-11678](https://issues.apache.org/jira/browse/YARN-11678). It's also called from the CGroupsResourceCalculator, which will handle the v1/v2 differences separately. 

It's called from OCIContainerRuntime, but that class doesn't use it cgroupHandler, DockerLinuxContainerRuntime calls it to check resource limits, and RuncContainerRuntime calls it for checking and storing cpu limits. Having this return the v1 or v2 handler based on the config will result in issues, even in mixed mode, where for example the CPU is mounted in a v1 structure, so I suggest leaving this as is for now, and updating it once DockerLinuxContainerRuntime and RuncContainerRuntime is updated to handle the v2 files.

Reply:
Sure, I changed it to return v1 handler only.